dipy/reconst/ivim.py

```diff
@@ -509,6 +536,360 @@ def _leastsq(self, data, x0):
+            default : 1e-15
+
+        maxiter: int, optional
+            Maximum number of iterations for the Differential Evolution in
+            SciPy.
+            default : 10
+
+        References
+        ----------
+        .. [1] Le Bihan, Denis, et al. "Separation of diffusion and perfusion
+               in intravoxel incoherent motion MR imaging." Radiology 168.2
+               (1988): 497-505.
+        .. [2] Federau, Christian, et al. "Quantitative measurement of brain
+               perfusion with intravoxel incoherent motion MR imaging."
+               Radiology 265.3 (2012): 874-881.
+        .. [3] Fadnavis, Shreyas et.al. "MicroLearn: Framework for machine
+               learning, reconstruction, optimization and microstructure
+               modeling, Proceedings of: International Society of Magnetic
+               Resonance in Medicine (ISMRM), Montreal, Canada, 2019.
+        """
+
+        self.maxiter = kwargs.get('maxiter', 10)
+        self.xtol = kwargs.get('xtol', 1e-8)
+        self.bvals = gtab.bvals
+        self.yhat_perfusion = np.zeros(self.bvals.shape[0])
+        self.yhat_diffusion = np.zeros(self.bvals.shape[0])
+        self.exp_phi1 = np.zeros((self.bvals.shape[0], 2))
+
+    @multi_voxel_fit
+    def fit(self, data):
+        r""" Fit method of the IvimModelVP model class
+
+        MicroLearn framework (VarPro)[1]_.
+
+        The VarPro computes the IVIM parameters using the MIX approach.
+        This algorithm uses three different optimizers. It starts with a
+        differential evolution algorithm and fits the parameters in the
+        power of exponentials. Then the fitted parameters in the first step are
+        utilized to make a linear convex problem. Using a convex optimization,
+        the volume fractions are determined. Then the last step is non linear
+        least square fitting on all the parameters. The results of the first
+        and second step are utilized as the initial values for the last step
+        of the algorithm. (see [1]_ and [2]_ for a comparison and a through
+        discussion).
+
+        References
+        ----------
+        .. [1] Fadnavis, Shreyas et.al. "MicroLearn: Framework for machine
+               learning, reconstruction, optimization and microstructure
+               modeling, Proceedings of: International Society of Magnetic
+               Resonance in Medicine (ISMRM), Montreal, Canada, 2019.
+        .. [2] Farooq, Hamza, et al. "Microstructure Imaging of Crossing (MIX)
+               White Matter Fibers from diffusion MRI." Scientific reports 6
+               (2016).
+
+        """
+        if SCIPY_LESS_0_15:
+            raise ValueError("Using the Variable Projection Method for " +
+                             "fitting needs SciPy >= 0.15.1")
+        else:
+            from scipy.optimize import differential_evolution
+
+        data_max = data.max()
+        data = data / data_max
+        data[data > 1] = 1
```

Comment:
Does this actually ever happen? I thought you just normalized by dividing by the maximum

Reply:
Will take it down! I didn't realize that it was still there :+1: 
Really good catch!